src/main/java/org/terasology/projsndwv/genetics/Genome.java

```diff
@@ -0,0 +1,151 @@
+            mutations.add(new HashMap<>());
+        }
+    }
+
+    public Iterator<GeneticsComponent> combine(GeneticsComponent g0, GeneticsComponent g1) {
+        if (!g0.isValid() || !g1.isValid()) {
+            return null;
+        }
+        else if (g0.size != size || g1.size != size) {
+            return null;
+        }
+
+        List<Mutation> possibleMutations = new ArrayList<>();
+
+        for (int i = 0; i < size; i++) {
+            Map<AllelePair, List<Mutation>> mutationsForLocus = mutations.get(i);
+            AllelePair ap = new AllelePair(g0.activeGenes.get(i), g1.activeGenes.get(i)); // TODO: Remove dependence on active genes
+            if (mutationsForLocus.containsKey(ap)) {
+                possibleMutations.addAll(mutationsForLocus.get(ap));
+            }
+        }
+
+        return new Iterator<GeneticsComponent>() {
+            @Override
+            public boolean hasNext() {
+                return true;
+            }
+
+            @Override
+            public GeneticsComponent next() {
+                GeneticsComponent result = new GeneticsComponent(size);
+
+                GeneticsComponent g0mutated = g0;
+                GeneticsComponent g1mutated = g1;
+
+                Collections.shuffle(possibleMutations);
+
+                for (Mutation mutation : possibleMutations) {
+                    if (random.nextFloat() < mutation.chance) {
+                        g0mutated = mutation.component;
+                        break;
+                    }
+                }
+
+                Collections.shuffle(possibleMutations);
+
+                for (Mutation mutation : possibleMutations) {
+                    if (random.nextFloat() < mutation.chance) {
+                        g1mutated = mutation.component;
+                        break;
+                    }
+                }
+
+                for (int i = 0; i < size; i++) {
+                    result.activeGenes.add(random.nextBoolean() ? g0mutated.activeGenes.get(i) : g0mutated.inactiveGenes.get(i));
+                    result.inactiveGenes.add(random.nextBoolean() ? g1mutated.activeGenes.get(i) : g1mutated.inactiveGenes.get(i));
+                }
+
+                return result;
+            }
+        };
+    }
+
+    // TODO: GeneticsComponents should probably consist of two separated genomes. This will allow for non-binary genetics and less data in these overrides later on.
+    public void registerMutation(int locus, int allele0, int allele1, GeneticsComponent override, float chance) {
+        AllelePair ap = new AllelePair(allele0, allele1);
+        Map<AllelePair, List<Mutation>> mutationsForLocus = mutations.get(locus);
+        List<Mutation> mutationList = mutationsForLocus.get(ap);
+        if (mutationList != null) {
+            mutationList.add(new Mutation(chance, override));
+        }
+        else {
+            mutationsForLocus.put(ap, new ArrayList<>(Collections.singletonList(new Mutation(chance, override))));
```

Comment:
Should this be a singleton list? Can you guarantee that it won't need to be more than a singleton?

Reply:
The list isn't a singleton. It's a list with the contents of a singleton added to it. i.e., the shortest possible way to initialize a list with one element. Because Java.